code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/unit/DefaultDeploymentUnitInstance.java

```diff
@@ -330,7 +331,19 @@ protected ServiceIndex createServiceIndex() {
                     InstanceConstants.FIELD_REQUESTED_IP_ADDRESS);
             String requestedIp = null;
             if (requestedIpObj != null) {
-                requestedIp = requestedIpObj.toString();
+                List<String> availableIPs = new ArrayList<>();
+                if (requestedIpObj instanceof List<?>) {
+                    for (Object item : (List<?>) requestedIpObj) {
+                        availableIPs.add(ObjectUtils.toString(item, null));
+                    }
+                }
+                List<String> usedIPs = context.serviceDao.getUsedIPsFromServiceIndexByServiceID(service);
```

Comment:
I guess i may still need to make this change: if ip is not available in poolManager, continue on the next available ip.